### PR TITLE
feat(plugin-meetings): optimisation for calls to wbxappapi

### PR DESCRIPTION
**Pull request by chburket**
# COMPLETES

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-288439

## This pull request addresses

In a few specific cases, this will avoid redirects when fetching meeting-info, and instead will request directly from convergedats, etc.

## by making the following changes

For SIP_URI and certain LOCUS_ID types, this uses the webex site from the sipUri to build the meeting-info uri.
Added two new utility functions to help with the parsing/building of uri.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

Added tests to cover the new functions. Manually tested with the web client to ensure it is working as expected.

### I certified that

- [X] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [X] I discussed changes with code owners prior to submitting this pull request

- [X] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 1]
```diff
@@ -264,4 +264,40 @@ MeetingInfoUtil.getRequestBody = (options) => {
   return body;
 };
 
+MeetingInfoUtil.getWebexSite = (uri) => {
+  const exceptedDomains = ['meet.webex.com', 'meetup.webex.com', 'ciscospark.com'];
+  const site = uri?.match(/.+@(.+\..+\.*.)$/)?.[1];
```
Comment (chburket) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:269]: One of these lines should change.  Right now, my regex only matches URIs with 2 periods (e.g. cisco.webex.com) but the exceptedDomains I pulled from SCF includes 'ciscospark.com'.  Do we expect to ever see a domain without the cisco/www/etc?  I should either remove ciscospark.com or change the regex to match similar URIs.

Comment (chburket) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:274]: This function is pretty limited in that it only handles URIs that look like '10019857020@convergedats.webex.com'. Will any sipUri's have a different format?

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js [hunk 2]
```diff
@@ -264,4 +264,45 @@ MeetingInfoUtil.getRequestBody = (options) => {
...
+
+  let preferredWebexSite = null;
+
+  switch (type) {
+    case _SIP_URI_:
+      preferredWebexSite = MeetingInfoUtil.getWebexSite(destination);
+      break;
+    case _LOCUS_ID_:
+      if (destination.info?.sipUri) {
+        preferredWebexSite = MeetingInfoUtil.getWebexSite(destination.info?.sipUri);
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:299]: info already has a webexSite info in it , We will not need the parsing logic here

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 1 above)
Comment (arun3528) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:274]: This is just optimization if we cannot match it appapi will redirect no issues

But ciscospark.com is not a webex site so we should neglect it 

Can you show where do you get ciscospark.com from

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 2 above)
Comment (chburket) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:299]: I don't understand.  distination.info.sipURL looks something like "10019857020@convergedats.webex.com" and I want it stripped down to "convergedats.webex.com".

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 1 above)
Comment (chburket) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:274]: https://sqbu-github.cisco.com/WebExSquared/spark-client-framework/blob/243e66ada181cd4ab94fae5b4053f4ce04997127/spark-client-framework/Services/WebexMeetingService/WebexMeetingService.cpp#L795

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 2 above)
Comment (chburket) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:299]: Oh, sorry, I see the webExSite property now.  Will change.

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js (hunk 1 above)
Comment (arun3528) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js:274]: i think its going to go away. we wont need it

**Review by arun3528 — APPROVED** [review]

**Pull request merged**